examples/ebi-example.js

```diff
@@ -0,0 +1,35 @@
+const { contentsSearch } = require('../lib/ebi');
+
+// NOTE: Assumes you have `GITHUB_PERSONAL_ACCESS_TOKEN` environment variable set
+const { GITHUB_PERSONAL_ACCESS_TOKEN } = process.env;
+
+const repos = [
+	'Financial-Times/ebi',
+	'Financial-Times/tako',
+	'Financial-Times/not-found-error'
+];
+
+async function main() {
+	const allResultsPromises = await contentsSearch({
+		token: GITHUB_PERSONAL_ACCESS_TOKEN,
+		filepath: 'package.json',
+		search: 'ebi'
+	})(repos);
+	const allResults = await Promise.all(
+		allResultsPromises.map(promise => {
+			return promise.catch(e => e);
```

Comment:
Should `ebi` search errors even `Promise.reject`? Do users care?

Reply:
The cli `--json` equivalent returns as `type: error` and is treated like search `match` and `no-match`

Reply:
what kinds of errors would these be?

Reply:
Invalid repo not found, GitHub connection problem, etc.

Reply:
Returning both synchronous and asynchronous results, where the async version throws errors, but the synchronous version doesn't throw. The synchronous version returns errors in objects with `type: 'error'` in a `searchErrors` array.